CalendarApp/GoogleCalendarService/BookingService.cs

```diff
@@ -78,7 +94,7 @@ public Event SearchEventById(string calendarId, string eventId)
         {
             try
             {
-                return _service.Events.List(calendarId).Execute().Items.SingleOrDefault(ev => ev.Id == eventId);
+                return GetEvents(calendarId).SingleOrDefault(ev => ev.Id == eventId);
```

Comment:
Maybe there is an API to search by ID. We could use  it to avoid loading all events to get a single one